### PR TITLE
Implement dpnp.ix_()

### PR DESCRIPTION
**Pull request by npolina4**
Implement dpnp.ix_(). Add tests.

- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: tests/test_indexing.py [hunk 1]
```diff
@@ -1057,3 +1057,35 @@ def test_fill_diagonal_error():
     arr = dpnp.ones((1, 2, 3))
     with pytest.raises(ValueError):
         dpnp.fill_diagonal(arr, 5)
+
+
+@pytest.mark.parametrize(
```
Comment (vtavana) [inline tests/test_indexing.py:1062]: Might be better to put the tests in a class, to be more organized.

File: tests/test_sycl_queue.py [hunk 2]
```diff
@@ -2504,3 +2504,21 @@ def test_ediff1d(device, to_end, to_begin):
...
+@pytest.mark.parametrize(
+    "device_1",
+    valid_devices,
+    ids=[device.filter_string for device in valid_devices],
+)
+def test_indices_sparse(device_0, device_1):
+    x0 = dpnp.array([0, 1], device=device_0)
+    x1 = dpnp.array([2, 4], device=device_1)
+    ixgrid = dpnp.ix_(x0, x1)
+    assert_sycl_queue_equal(ixgrid[0].sycl_queue, x0.sycl_queue)
```
Comment (vtavana) [inline tests/test_sycl_queue.py:2523]: Although this test works, but having output on different devices does not allow to re-construct an array.

It is kind of similar to the discussion about data type of input sequences. Although the function works for all data types, but for its intended use case of this function, it only make sense to have integer/boolean dtypes. 
I do not think we need to change anything but it is good to be aware of it.

```
import dpnp
x0 = dpnp.array([0, 1], device="cpu")
x1 = dpnp.array([2, 4], device="gpu")
ixgrid = dpnp.ix_(x0, x1)

a = dpnp.arange(10).reshape(2, 5)
a[ixgrid]
# ExecutionPlacementError: ...
```

File: dpnp/dpnp_iface_indexing.py [hunk 3]
```diff
@@ -897,6 +898,86 @@ def indices(
...
+    N dimensions.
+
+    Using `ix_` one can quickly construct index arrays that will index
+    the cross product. ``a[dpnp.ix_([1,3],[2,5])]`` returns the array
+    ``[[a[1,2] a[1,5]], [a[3,2] a[3,5]]]``.
+
+    Parameters
+    ----------
+    x1, x2,..., xn : {dpnp.ndarray, usm_ndarray}
+        1-D sequences. Each sequence should be of integer or boolean type.
```
[the remark this answers is not in the document]
Comment (npolina4) [reply dpnp/dpnp_iface_indexing.py:916]: NumPy does not provide any further explanation.

File: dpnp/dpnp_iface_indexing.py [hunk 4]
```diff
@@ -897,6 +898,86 @@ def indices(
...
+    >>> a = np.arange(10).reshape(2, 5)
+    >>> a
+    array([[0, 1, 2, 3, 4],
+           [5, 6, 7, 8, 9]])
+    >>> x1 = np.array([0, 1])
+    >>> x2 = np.array([2, 4])
+    >>> ixgrid = np.ix_(x1, x2)
+    >>> ixgrid
+    (array([[0],
+           [1]]), array([[2, 4]]))
```
[the remark this answers is not in the document]
Comment (npolina4) [reply dpnp/dpnp_iface_indexing.py:939]: This is the original output

**Review by vtavana — APPROVED** [review]
Thank you, @npolina4!

**Pull request merged**